### PR TITLE
FOCUS #532: PricingCategory update - Classify commitment discount purchase records as Standard instead of Committed

### PR DESCRIPTION
**Pull request by ijurica**
**Lead Maintainer**: Irena

Classify commitment discount purchase records as Standard instead of Committed, by rephrasing the following requirement:

> * PricingCategory MUST be "Committed" when [CommitmentDiscountId](#commitmentdiscountid) is not null.

New version:

> * PricingCategory MUST be "Committed" when [CommitmentDiscountId](#commitmentdiscountid) is not null, except when ChargeCategory is "Purchase" and the [ResourceId](#resourceid) matches the CommitmentDiscountId.

or 

> * PricingCategory MUST be "Committed" when [CommitmentDiscountId](#commitmentdiscountid) is not null, except when ChargeCategory is "Purchase" and the [ResourceId](#resourceid) matches the CommitmentDiscountId, representing a charge related to the purchase of that commitment discount.

### CODE REVIEW

File: specification/columns/pricingcategory.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ The PricingCategory column adheres to the following requirements:
 * PricingCategory MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory.
 * PricingCategory MUST be one of the allowed values.
 * PricingCategory MUST be "Standard" when pricing is predetermined at the agreed upon rate for the [billing account](#glossary:billing-account).
-* PricingCategory MUST be "Committed" when [CommitmentDiscountId](#commitmentdiscountid) is not null.
+* PricingCategory MUST be "Committed" when [CommitmentDiscountId](#commitmentdiscountid) is not null, except when ChargeCategory is "Purchase" and the [ResourceId](#resourceid) matches the CommitmentDiscountId.
```
Comment (ijurica) [inline specification/columns/pricingcategory.md:11]: option B: 

```suggestion
 * PricingCategory MUST be "Committed" when [CommitmentDiscountId](#commitmentdiscountid) is not null, except when ChargeCategory is "Purchase" and the [ResourceId](#resourceid) matches the CommitmentDiscountId, representing a charge related to the purchase of that commitment discount.
```

Comment (flanakin) [reply specification/columns/pricingcategory.md:11]: Option C: Leave this open for purchase commitments (e.g., consulting hours) and remove the relationship with the CommitmentDiscountId. This is the same language in CommitmentDiscountId without binding us to that column.
```suggestion
* PricingCategory MUST be "Committed" when the charge is related to a *commitment discount*.
```

Comment (ijurica) [reply specification/columns/pricingcategory.md:11]: #### PricingCategory values:

> **Standard:** Charges priced at the agreed upon rate for the billing account, including negotiated discounts. This includes any flat rate and volume/tiered pricing but does not include dynamic or commitment discount pricing.
>
> **Committed:** Charges with reduced prices due to a commitment discount specified by the Commitment Discount ID.

#### DiscountHandling requirement:

> The CommitmentDiscountId and ResourceId MUST be set to the ID assigned to the *commitment discount*. ChargeCategory MUST be set to "Purchase" on rows that represent a purchase of a *commitment discount*.

#### Question:

A purchase of a commitment is related to a commitment discount, but should it be it classified as "Committed" (and therefore covered by that particular commitment) or as "Standard"?

#### Simplified dataset

| Version   | ChargeDescription | ChargeCategory | Pricing<br>Category | Commitment<br>DiscountId | ResourceId |
|-----------|-------------------|----------------|-----------------|-----------------------|------------|
| Current & Option C   | CD-ID-1 purchase  | Purchase       | Committed       | CD-ID-1               | CD-ID-1    |
| Options A & B | CD-ID-1 purchase  | Purchase       | Standard        | CD-ID-1               | CD-ID-1    |

Comment (cnharris10) [inline specification/columns/pricingcategory.md:11]: Option D?
```suggestion
* PricingCategory MUST be "Committed" when [CommitmentDiscountId](#commitmentdiscountid) is not null and ChargeCategory is "Usage".
```

Comment (ijurica) [reply specification/columns/pricingcategory.md:11]: Are we absolutely certain that Commitment discounts, including any future ones, cannot be applied to purchase-related records?

Comment (cnharris10) [reply specification/columns/pricingcategory.md:11]: Yes, 💯. A commitment discount's reduction only applies to usage rows. Any additional reduction to the purchase is a negotiated discount.

**Comment by jpradocueva** [comment]
### Action Items 
- [ ] [Maintainers-[#532](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/532)/[#533](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/533)] Irena, @ijurica : To contact Chris, @cnharris10  and Michael, @flanakin  separately to get their feedback on the issue.
- [ ] [Maintainers-[#532](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/532)/[#533](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/533)] Irena, @ijurica : Offline Resolution. The issue will be resolved offline once Chris and Michael provide their input. The final resolution will be documented in the PR to ensure transparency.

**Comment by jpradocueva** [comment]
### Action Items from TF-2 call on Sep 4:
- [ ] [TF2-[#533](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/533)] Irena, @ijurica: Update the PR to reflect this change before the next review.

**Review by ahullah — APPROVED** [review]

File: specification/columns/pricingcategory.md (hunk 1 above)
Comment (flanakin) [reply specification/columns/pricingcategory.md:11]: I'm not sure that's a fair assumption. That's just how it works today for the major cloud providers. We don't know if anyone's using commitment discounts for purchases. I don't think we should limit commitment discounts to usage only. As an example, a domain registrar may offer commitment discounts for organizations who purchase 1,000 domains, which would be a bulk of purchases and not usage. These are valid discounts for committing to a specific quantity of purchases. We should support that.

Comment (flanakin) [reply specification/columns/pricingcategory.md:11]: > A purchase of a commitment is related to a commitment discount
Good point. How about...
```suggestion
* PricingCategory MUST be "Committed" when the charge is subject to an existing *commitment discount* and is not the purchase of a *commitment discount*.
```

**Comment by jpradocueva** [comment]
### Action Items from Members' call on Sep 5:
- [ ] [Members-[#533](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/533)] Irena, @ijurica : To finalize the PR based on group feedback.

File: specification/columns/pricingcategory.md (hunk 1 above)
Comment (cnharris10) [reply specification/columns/pricingcategory.md:11]: I'm fine with the proposed solution as is, so we can resolve this comment.

**Review by cnharris10 — APPROVED** [review]

**Comment by jpradocueva** [comment]
### Action Items from TF-3 call on Sep 6:
- [ ] [TF1-[#533](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/533)] Irena, @ijurica : Prepare the final version of the use case document by Monday for Task Force 1 review.
- [ ] [TF1-[#533](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/533)] Riley, @rileyjenk : Review Google Spreadsheet and Google Doc early next week and provide feedback before Thursday's meeting.
- [ ] [TF1-[#533](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/533)] Chris, @cnharris10 : Present the updated classification in the Thursday members meeting for formal approval.

File: specification/columns/pricingcategory.md (hunk 1 above)
Comment (ijurica) [reply specification/columns/pricingcategory.md:11]: Moving forward with @flanakin's suggestion. Although I usually prefer precision in normative paragraphs (almost like instructions for those implementing validators, whenever applicable) in this case, it’s not just about readability but also about safety. 
I’d suggest making a small adjustment by replacing "a" with "the" to clarify that we are specifically referring to the purchase of that particular commitment discount.

File: specification/columns/pricingcategory.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ The PricingCategory column adheres to the following requirements:
 * PricingCategory MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory.
 * PricingCategory MUST be one of the allowed values.
 * PricingCategory MUST be "Standard" when pricing is predetermined at the agreed upon rate for the [billing account](#glossary:billing-account).
-* PricingCategory MUST be "Committed" when [CommitmentDiscountId](#commitmentdiscountid) is not null.
+* PricingCategory MUST be "Committed" when the charge is subject to an existing *commitment discount* and is not the purchase of a *commitment discount*.
```
Comment (ijurica) [inline specification/columns/pricingcategory.md:11]:
```suggestion
* PricingCategory MUST be "Committed" when the charge is subject to an existing *commitment discount* and is not the purchase of the *commitment discount*.
```

referring to the specific commitment discount that the charge is subject to

**Review by udam-f2 — APPROVED** [review]

**Comment by jpradocueva** [comment]
Approved by the Members on Sep 12 call.

**Comment by jpradocueva** [comment]
### Action Items from Members' call on Sep 12:
- [ ] [TF2-[#533](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/533)] Irena, @ijurica : Ensure the phrasing is updated accordingly.

**Pull request merged**